create.js

```diff
@@ -23,10 +23,25 @@ module.exports = function(_nativeProtocols) {
   return publicApi;
 
   function execute(options) {
-    var clientRequest;
     var fetchedUrls = [];
-
-    return (clientRequest = cb());
+    var clientRequest = cb();
+    var requestProxy = Object.create(clientRequest);
+
+    // separate the proxy's event handlers from the request's
+    requestProxy._events = {};
+    requestProxy._eventsCount = 0;
+    if (options.userCallback) {
+      requestProxy.on('response', options.userCallback);
+    }
+    // forward all request events on the proxy except `response`
```

Comment:
See my response to #30.

I think we should probably only emit events that are logical given that we are making redirection transparent. Also, we would ideally  provide a similar server side experience that `browserify-http`, `browserify-http-2` provide in the browser. I'm guessing many of the events simply aren't detectable in the browser. So I would definitely say just skip it.